src/pytorch_lightning/loggers/tensorboard.py

```diff
@@ -242,9 +242,11 @@ def log_graph(self, model: "pl.LightningModule", input_array: Optional[Tensor] =
             if input_array is not None:
                 input_array = model._on_before_batch_transfer(input_array)
                 input_array = model._apply_batch_transfer_handler(input_array)
-                model._running_torchscript = True
-                self.experiment.add_graph(model, input_array)
-                model._running_torchscript = False
+                pl.LightningModule._running_torchscript = True
+                try:
+                    self.experiment.add_graph(model, input_array)
```

Comment:
@carmocca should we go with a context manager? exit gets called when an exception occurs inside, I think it would look slightly cleaner.

Reply:
I would agree if this was something user facing, but defining a context manager class and instantiating it etc just to set this attribute seems overkill to me.

No strong opinion though, I can do it if you prefer

Reply:
Done